src/ua_types.c

```diff
@@ -114,6 +115,48 @@ UA_cleanupDataTypeWithCustom(const UA_DataTypeArray *customTypes) {
 /*****************/
 /* Builtin Types */
 /*****************/
+#define FORMAT_STACK_SIZE 256
+
+
+UA_StatusCode
+UA_String_fromFormatWithBuffer (UA_String *string, const char *format, ...)
+{
+    if(!string || !string->data || string->length < 1) return UA_STATUSCODE_BADINTERNALERROR;
+    va_list args;
+    va_start (args, format);
+    int ret = mp_vsnprintf((char *)string->data, string->length, format, args);
+    /* Len -1 to not include the null term in the string */
+    string->length = UA_MIN (string->length-1, (size_t) ret);
+    return UA_STATUSCODE_GOOD;
+}
+
+UA_String
+UA_String_fromFormat (const char *format, ...)
+{
+    char tmp[FORMAT_STACK_SIZE];
+    va_list args;
+    va_start (args, format);
+    int ret = mp_vsnprintf(tmp, FORMAT_STACK_SIZE, format, args);
+    va_end(args);
+    if (ret <= 0) return UA_STRING_NULL;
+    UA_String s;
+    s.length = ret;
+    if (ret < FORMAT_STACK_SIZE)
+    {
+        s.data = (UA_Byte*) UA_malloc(s.length);
```

Comment:
Check for malloc fails.
This can happen on embedded targets.

Otherwise great.